src/shared/generator.c

```diff
@@ -119,7 +123,7 @@ static int write_fsck_sysroot_service(
                 "DefaultDependencies=no\n"
                 "BindsTo=%4$s\n"
                 "Conflicts=shutdown.target\n"
-                "After=%5$s%6$slocal-fs-pre.target %4$s\n"
+                "After=%5$s%6$slocal-fs-pre.target %4$s %8$s\n"
```

Comment:
frankly, I think it should be the other way round: the uncommon component (mkfs) should carry the dep onto the common component (fsck).

i.e. please make this a Before= dep from the  mkfs unit

Reply:
@poettering That ends up being a touch trickier, because the code that generates the mkfs unit doesn't have a separate codepath for /sysroot to point it at systemd-fsck-root instead of systemd-fsck@DEV, whereas the fsck code does use a custom codepath. I'll make the change if you think it's worth it though.

Reply:
Actually, not tricky at all. Done.